pkg/volume/csi/csi_attacher_test.go

```diff
@@ -331,16 +332,47 @@ func TestAttacherDetach(t *testing.T) {
 		volID      string
 		attachID   string
 		shouldFail bool
+		reactor    func(action core.Action) (handled bool, ret runtime.Object, err error)
 	}{
 		{name: "normal test", volID: "vol-001", attachID: getAttachmentName("vol-001", testDriver, nodeName)},
 		{name: "normal test 2", volID: "vol-002", attachID: getAttachmentName("vol-002", testDriver, nodeName)},
-		{name: "object not found", volID: "vol-001", attachID: getAttachmentName("vol-002", testDriver, nodeName), shouldFail: true},
+		{name: "object not found", volID: "vol-non-existing", attachID: getAttachmentName("vol-003", testDriver, nodeName)},
```

Comment:
how is this test different from the longer one following? They both test when the attachment is not found no? One by just not creating the right attachment, and the other with this reactor. If they are they are the same, I would favor keeping the simpler one

Reply:
Sorry, I was referring to this test ""volume attachment not found -> success" and this test "object not found" both of which should return success and the tested functionality is a volume attachment that is not found.

Reply:
I see, I must have been blind... It's fixed now